netbox/core/plugins.py

```diff
@@ -0,0 +1,195 @@
+import datetime
+import importlib
+import importlib.util
+from dataclasses import dataclass, field
+from typing import Optional
+
+import requests
```

Comment:
@natm points out that the way we're using `requests` does not honor the system proxy settings, which might result in unnecessary catalog load failures and possibly also data leaks. Can we update the client to respect this setting?